lib/addons/coredns/index.ts

```diff
@@ -8,7 +8,8 @@ export class CoreDnsAddOn extends CoreAddOn {
     constructor(version?: string) {
         super({
             addOnName: "coredns",
-            version: version ?? "v1.8.4-eksbuild.1"
+            version: version ?? "v1.8.4-eksbuild.1",
+            saName: "coredns"
```

Comment:
not strictly necessary for anything other than EBS CSI add-on, since none of the rest pass the policies for the IRSA account.